### PR TITLE
Radio button

### PR DESCRIPTION
**Pull request by iwwwi**
## What does this PR do:

* fixes issue #17
* adds stateless radio button component.
* adds custom styles to  the radio button.
* adds aria properties for the radio button since we are not using the native html tag
* adds test
* adds md
* updates snapshots

## Where should the reviewer start:
* components/radioButton/radioButton.js
* components/radioButton/radioButton.scss
* components/radioButton/radioButton.md

screen shot 2017-02-23 at 11 22 22

### CODE REVIEW

File: components/radioButton/radioButton.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    <div className={className} {...restProps}>
+      <input
+        checked={checked}
+        disabled={disabled}
+        id={id}
+        name={name}
+        onChange={onChange}
+        type="radio"
+      />
+      <label aria-checked={checked} htmlFor={id} role="radio">
```
Comment (ibsukru) [inline components/radioButton/radioButton.js]: You don't need aria-checked={checked} | role="radio", unless you are using for='radioId'

Comment (iwwwi) [inline components/radioButton/radioButton.js]: Here we are using the label to also style the input element. which means we are loosing the accessibility properties from the native input element. In this case we need to add this props so that we can navigate as we have the native element (ex.: navigating with tabs and focusing the element).

File: components/radioButton/radioButton.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+};
+
+RadioButton.propTypes = {
+  /**
+   * Represents the state of the radio button.
+   */
+  checked: PropTypes.bool,
+
+  /**
+   * Radio Button value.
```
Comment (asci) [inline components/radioButton/radioButton.js]: Radio Button __label__

File: components/radioButton/radioButton.scss [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  font-family: $tx-generic-font-secondary-font-family, $tx-generic-font-secondary-generic-family;
+  font-size: 14px;
+  font-weight: $tx-generic-font-primary-weight-regular;
+  min-height: 20px;
+  padding-bottom: 2px;
+  padding-right: 10px;
+  padding-top: 2px;
+  position: relative;
+
+  & input[type="radio"] {
```
Comment (asci) [inline components/radioButton/radioButton.scss]: is this `input[type="radio"]`required?

File: components/radioButton/radioButton.js (hunk 1 above)
Comment (ibsukru) [inline components/radioButton/radioButton.js]: That is exactly what i mean :) you have label with for='radio' attribute already, aria-checked={checked} role="radio" is not required, so accessibility devices understand this is a label for radio input

**Comment by mAiNiNfEcTiOn** [comment]
@iwwwi you should `git rebase master` and `npm run update-snapshots`.

File: components/radioButton/radioButton.scss [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    left: -9999px;
+    position: absolute;
+
+    & + label {
```
Comment (mAiNiNfEcTiOn) [inline components/radioButton/radioButton.scss]: So you're saying that she should do:

```scss
.ui-price input[type="radio"] + label {
  // code here
}
```

?

Comment (ibsukru) [inline components/radioButton/radioButton.scss]: Actually .ui-price__label { ... 
adding that class is a solution

File: themes/_default.yaml [hunk 5]
```diff
@@ -40,7 +40,7 @@ generic:
     secondary-lighter: &secondary-lighter '#FFD97C'
 
     text-dark: &text-dark '#333'
-    text: &text '#666'
```
Comment (iwwwi) [inline themes/_default.yaml]: will undo the change :)

File: components/radioButton/radioButton.scss (hunk 4 above)
Comment (iwwwi) [inline components/radioButton/radioButton.scss]: @ibsukru how do you suggest to use the pseudo selectors using the  mentioned class? I am bit concerned because  i need to combine multiple elements with their pseudo selectors, for ex. `input[type="radio"]:checked + label:after`. In this case I will have to write something like `input[type="radio"]:checked + label.ui-radio__label:after`
